### PR TITLE
system-variables: Merge with tidb-specific-sysvars (#3152)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #3152 to release-4.0

---

### What is changed, added or deleted? (Required)

This PR proposes merging the system-variables and tidb-specific-system-variables documents together.

The motivation for this, is that they both have the same semantics. It is just a namespace that differs, but this can clearly be visible by sorting the variables in alphabetical order (something else I've done.)

In the process of creating this PR, I discovered that several sysvars are undocumented. I have created https://github.com/pingcap/docs/issues/3155 to handle separately.

Also, `SHARD_ROW_ID_BITS` was documented as a system variable but it is not. I have moved it to the statement reference for `CREATE TABLE` and `ALTER TABLE`.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

This is original work.

### CODE REVIEW

**Comment by yikeke** [comment]
Please help delete tidb-specific-system-variables.md. @nullnotnil (P.S. Cherry-picker bot can't delete files because of a known bug)

File: system-variables.md [hunk 1]
```diff
@@ -1,129 +1,842 @@
...
-title: The System Variables
-summary: Learn how to use the system variables in TiDB.
+title: System Variables
+summary: Use system variables to optimize performance or alter running behavior.
 category: reference
+<<<<<<< HEAD
 aliases: ['/docs/stable/system-variables/','/docs/v4.0/system-variables/','/docs/stable/reference/configuration/tidb-server/mysql-variables/']
+=======
+aliases: ['/docs/dev/system-variables/','/docs/dev/reference/configuration/tidb-server/mysql-variables/', '/docs/dev/tidb-specific-system-variables/','/docs/dev/reference/configuration/tidb-server/tidb-specific-variables/']
+>>>>>>> 7c3206d... system-variables: Merge with tidb-specific-sysvars (#3152)
```
Comment (yikeke) [inline system-variables.md:9]:
```suggestion
<<<<<<< HEAD
aliases: ['/docs/stable/system-variables/','/docs/v4.0/system-variables/','/docs/stable/reference/configuration/tidb-server/mysql-variables/']
=======
aliases: ['/docs/stable/system-variables/','/docs/v4.0/system-variables/','/docs/stable/reference/configuration/tidb-server/mysql-variables/','/docs/stable/tidb-specific-system-variables/','/docs/v4.0/tidb-specific-system-variables/','/docs/stable/reference/configuration/tidb-server/tidb-specific-variables/']
```

**Comment by nullnotnil** [comment]
@yikeke `tidb-specific-system-variables.md` has been deleted, thanks for the heads up. This is ready for review.

File: system-variables.md [hunk 2]
```diff
@@ -1,129 +1,838 @@
 ---
-title: The System Variables
-summary: Learn how to use the system variables in TiDB.
+title: System Variables
+summary: Use system variables to optimize performance or alter running behavior.
 category: reference
-aliases: ['/docs/stable/system-variables/','/docs/v4.0/system-variables/','/docs/stable/reference/configuration/tidb-server/mysql-variables/']
+aliases: ['/docs/stable/system-variables/','/docs/v4.0/system-variables/','/docs/stable/reference/configuration/tidb-server/mysql-variables/','/docs/stable/tidb-specific-system-variables/','/docs/v4.0/tidb-specific-system-variables/','/docs/stable/reference/configuration/tidb-server/tidb-specific-variables/']
```
Comment (yikeke) [inline system-variables.md:5]:
```suggestion
aliases: ['/tidb/stable/tidb-specific-system-variables','/docs/stable/system-variables/','/docs/v4.0/system-variables/','/docs/stable/reference/configuration/tidb-server/mysql-variables/','/docs/stable/tidb-specific-system-variables/','/docs/v4.0/tidb-specific-system-variables/','/docs/stable/reference/configuration/tidb-server/tidb-specific-variables/']
```
I just realized we forgot this alias: '/tidb/stable/tidb-specific-system-variables'. It is to redirect the deleted file to system-variables.md so no dead link will be caused by the deletion. @nullnotnil

Comment (yikeke) [reply system-variables.md:5]: I'll add this `'/tidb/dev/tidb-specific-system-variables'` alias in [my PR](https://github.com/pingcap/docs/pull/3239).

Comment (yikeke) [reply system-variables.md:5]: Added it in https://github.com/pingcap/docs/pull/3239/commits/1c5608e2117f48f0c2ba81ed7470948ecc9997cb

**Pull request merged**